central/debug/service/diagnostics.go

```diff
@@ -276,8 +276,10 @@ func (s *serviceImpl) getK8sDiagnostics(ctx context.Context, zipWriter *zip.Writ
 	// Pull data from the central cluster.
 	// TODO: It would be nice if we could add a flag to a sensor connection that indicates whether this sensor is
 	// running colocated with central, so we could skip this.
-	wg.Add(1)
-	go pullCentralClusterDiagnostics(subCtx, filesC, &wg, opts.since)
+	if opts.withCentral {
+		wg.Add(1)
+		go pullCentralClusterDiagnostics(subCtx, filesC, &wg, opts.since)
```

Comment:
For my own understanding: does this function collect kubernetes manifests for the diagnostic bundle? 
The code in L281 fetches Central resources, the above fetches secured cluster k8s resources from all connected sensors?

Reply:
> does this function collect kubernetes manifests for the diagnostic bundle?
Yes for stackrox namespace. Manifests, logs and events. Also the root ca configmap.

> The code in L281 fetches Central resources, the above fetches secured cluster k8s resources from all connected sensors?

Correct. This is done in addition to the secured clusters in case central is deployed without sensor.

That's why the caveat exists that if central has sensor this itself isn't enough to hide all central data. For ACSMS, since central doesn't have sensor, it works out.

Reply:
Thank you @theencee!